### PR TITLE
Add optional page size increase to pulls list

### PR DESCRIPTION
**Pull request by thieman**
Redirecting from #58 to the `next` branch, sorry. Some existing discussion there.

### CODE REVIEW

**Comment by sigma** [comment]
could we have 100 as a custom variable instead of hard-coded? that might help a bit if someday github decides to change the limit, or for other reasons I guess

**Comment by thieman** [comment]
I'll take a stab at that, yeah. Does the other part with the query nulling make sense?

File: gh-api.el [hunk 1]
```diff
@@ -158,6 +158,7 @@
       (let ((req (oref resp :-req)))
         (oset resp :data-received nil)
         (oset req :url next)
+        (oset req :query nil)
```
Comment (thieman) [inline gh-api.el]: Copying comment from previous PR

Here we already get a fully-formed URL from the link header, which from what I can tell includes any relevant query args from your first request such as per_page. We do not want to apply any of our own query args to it from our first request. So we null out :query here to just use the Link that GitHub gave us.

Comment (sigma) [inline gh-api.el]: so I guess *ideally* we'd parse the link, split it in url/query, and use those parts in the followup request.
Maybe it's easy enough that we can do that right now?

**Comment by thieman** [comment]
@sigma Updated per your suggestions. I'll make a comment or two inline.

File: gh-url.el [hunk 2]
```diff
@@ -152,8 +152,10 @@
            (gh-url-run-request req resp)))))))
 
 (defun gh-url-form-encode (form)
-  (mapconcat (lambda (x) (format "%s=%s" (car x) (cdr x)))
-             form "&"))
+  (if (stringp form)
+      form
+    (mapconcat (lambda (x) (format "%s=%s" (car x) (cdr x)))
+               form "&")))
```
Comment (thieman) [inline gh-url.el]: This looks hacky at first glance but it seemed like a better idea than going through all the effort to change the string to a list of dot pairs just to deconstruct it later on. Any other ideas?

File: gh-api.el [hunk 3]
```diff
@@ -155,9 +155,13 @@
     (call-next-method)
     (oset resp :data (append previous-data (oref resp :data)))
     (when (and next (not (equal 304 (oref resp :http-status))))
-      (let ((req (oref resp :-req)))
+      (let* ((req (oref resp :-req))
```
Comment (sigma) [inline gh-api.el]: maybe using `url-generic-parse-url` would be more elegant?

Comment (thieman) [inline gh-api.el]: Seems likely, thanks for the suggestion

**Pull request closed without merging**